src/java/htsjdk/samtools/util/CloseableIterator.java

```diff
@@ -40,6 +46,19 @@
  * We do not inherit from java.io.Closeable because IOExceptions are a pain to deal with.
  */
 public interface CloseableIterator<T> extends Iterator<T>, Closeable {
+    /** Should be implemented to close/release any underlying resources. */
+    void close();
 
-    public void close();
+    /** Consumes the contents of the iterator and returns it as a List. */
+    default List<T> toList() {
+        final List<T> list = new ArrayList<>();
+        while (hasNext()) list.add(next());
+        return list;
+    }
+
+    /** Returns a Stream that will consume from the underlying iterator. */
+    default Stream<T> stream() {
+        final Spliterator<T> s = Spliterators.spliteratorUnknownSize(this, Spliterator.ORDERED);
```

Comment:
I agree it would be nice, but I'm not sure how possible it is without a bunch of wrapper-ing of the iterators.  Let me see if I can come up with something that's not too ugly.